Init commit.

jherrmann06 opened the pull request:
Code is not ready yet, but it is as complete as it can be without a swerve chassis ready.

jherrmann06 committed:
Init commit. Code is not ready yet, but it is as complete as it can be without a swerve chassis ready.

jdlanning commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -0,0 +1,50 @@
+     */
+    public static final double DRIVETRAIN_WHEELBASE_METERS = 0.601; //Mesasured in cad; FIXME Measure and set wheelbase
+
+    public static final int DRIVETRAIN_PIGEON_ID = 0; // FIXME Set Pigeon ID
```
We can delete the constants we arnt using

jdlanning commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,93 @@
+  // The robot's subsystems and commands are defined here...
+  private final DrivetrainSubsystem m_drivetrainSubsystem = new DrivetrainSubsystem();
+
+  private final XboxController m_controller = new XboxController(0);
```
Lets add a constant for 0

jdlanning commented on src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:
```diff
@@ -0,0 +1,196 @@
+  // The important thing about how you configure your gyroscope is that rotating the robot counter-clockwise should
+  // cause the angle reading to increase until it wraps back over to zero.
+  // FIXME Remove if you are using a Pigeon
+  private final PigeonIMU m_pigeon = new PigeonIMU(DRIVETRAIN_PIGEON_ID);
```
We can remove the Pigeon

jdlanning commented on src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:
```diff
@@ -0,0 +1,196 @@
+    //   Your module has a Falcon 500 and a NEO on it. The Falcon 500 is for driving and the NEO is for steering.
+    //
+    // Mk3SwerveModuleHelper.createNeoFalcon500(...)
+    //   Your module has a NEO and a Falcon 500 on it. The NEO is for driving and the Falcon 500 is for steering.
```
We can do a little clean up here as well for the commented code while we wait on the chassis to be ready.

jdlanning commented on src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:
```diff
@@ -0,0 +1,165 @@
+            BACK_LEFT_MODULE_STEER_OFFSET
+    );
+
+    m_backRightModule = Mk3SwerveModuleHelper.createNeo(
```
I believe we have the Mk4 modules. @smitty1707 can confirm.

ASmith1675 commented:
Yes. Mk4 modules. It's possible that the code is the same as for mk3 I
think there were just mechanical improvements.

On Thu, Dec 1, 2022, 8:50 PM Justin Lanning ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java
> <https://github.com/frc1675/swerve-dev/pull/1#discussion_r1037739948>:
>
> > +            FRONT_RIGHT_MODULE_STEER_ENCODER,
> +            FRONT_RIGHT_MODULE_STEER_OFFSET
> +    );
> +
> +    m_backLeftModule = Mk3SwerveModuleHelper.createNeo(
> +            tab.getLayout("Back Left Module", BuiltInLayouts.kList)
> +                    .withSize(2, 4)
> +                    .withPosition(4, 0),
> +            Mk3SwerveModuleHelper.GearRatio.STANDARD,
> +            BACK_LEFT_MODULE_DRIVE_MOTOR,
> +            BACK_LEFT_MODULE_STEER_MOTOR,
> +            BACK_LEFT_MODULE_STEER_ENCODER,
> +            BACK_LEFT_MODULE_STEER_OFFSET
> +    );
> +
> +    m_backRightModule = Mk3SwerveModuleHelper.createNeo(
>
> I believe we have the Mk4 modules. @smitty1707
> <https://github.com/smitty1707> can confirm.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/frc1675/swerve-dev/pull/1#pullrequestreview-1201910865>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AF7WAHZFAYTHR45PCKMF3TDWLFPYXANCNFSM6AAAAAASMCK2J4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

jherrmann06 committed:
Updated .gitignore to ignore garbage files

jherrmann06 committed:
Responded to code review feedback, cleaned up comments

jherrmann06 committed:
Updated project to WPILib version 2022

Had to install new vendor libraries, update also triggered changes to gradle files

jherrmann06 committed:
Updated imports to reflect lib changes

jherrmann06 committed:
Now reflects lib changes, has Mk4 swerve modules

jherrmann06 committed:
Updated constants.java to include correct CAN ids

jdlanning commented on src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java:
```diff
@@ -1,18 +1,35 @@
+import static frc.robot.Constants.FRONT_RIGHT_MODULE_DRIVE_MOTOR;
+import static frc.robot.Constants.FRONT_RIGHT_MODULE_STEER_ENCODER;
+import static frc.robot.Constants.FRONT_RIGHT_MODULE_STEER_MOTOR;
+import static frc.robot.Constants.FRONT_RIGHT_MODULE_STEER_OFFSET;
```
Overall looks good. Only thing that caught my eye was these imports looks a little verbose (separate import for each constant seems overkill). It should be possible to just import all constants in one line

jdlanning reviewed (approved):
My one comment is very minor so just keep it in mind if we end up doing more cleanup . I will merge so we can start making smaller PRs

jherrmann06 committed:
Removed unnecessary comments

jherrmann06 committed:
Measured and set offsets

jherrmann06 committed:
Re-measured and updated offsets

jherrmann06 committed:
Removed unnecessary println statement

jherrmann06 committed:
Added scaling control to robot speed and each axis
Scaler values are provided by 4 shuffleboard widgets
